### ISSUE TITLE
r509 does not work with ruby 2.4

### ISSUE THREAD

**Issue by will** [r509/r509#122]
Tried upgrading one of my projects to 2.4 and ran into problems with r509, so I tried to run the specs in the project to see what was up. I'm not sure what the root issue is, but about half the specs fail on 2.4. There didn't appear to be any open issues on this, so I figured I'd open a ticket. Thanks!

```
r509 ➤ rbenv local 2.3.3                                                                                                                                                                        git:master*
r509 ➤ rspec -fp                                                                                                                                                                                git:master*
.......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Finished in 3.12 seconds (files took 2.08 seconds to load)
1255 examples, 0 failures

Coverage report generated for RSpec to /private/tmp/r509/coverage. 6486 / 6505 LOC (99.71%) covered.
r509 ➤ rbenv local 2.4.0
r509 ➤ rspec -fp      
F.................................................................................FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF...FFFFFFFF.....FFFF.FFFF......FFFF.F.FFFF......FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF......FFFFFFFFFFFFF.FFFFFFFFFFF....FFFFFFFFFFFFFFFFFF....FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF..FFFFFFFFFF....FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF.....FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF..............FFFFFFFFFFFFFFFFFFFFFF..FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF........................FFFFF.........FFFFFFFFFFFFFFF...........FF.F..................F.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF....................F.......F.............F............F........FFF.............................................................................FFF.F...........FFFF.FFF..F/private/tmp/r509/lib/r509/private_key.rb:131: warning: constant OpenSSL::Cipher::Cipher is deprecated
....FFFFFFFFF.FF..F.................FF...........................FFFFFFFFFFFFFFF....................................................................F...F...F...F.......F..................................................................

…
Finished in 2.75 seconds (files took 0.84809 seconds to load)
1255 examples, 692 failures
…
```

**Comment by fdr** [comment]
This is probably connected with the Ruby 2.4 switch to OpenSSL v1.1.0.

**Comment by reaperhulk** [comment]
Makes sense! Hopefully someone can step up to help fix it!

**Comment by reaperhulk** [comment]
Resolved in #124
